lib/core/src/Cardano/Wallet.hs

```diff
@@ -3432,16 +3432,15 @@ data ErrBalanceTx
     | ErrBalanceTxSelectAssets ErrSelectAssets
     | ErrBalanceTxMaxSizeLimitExceeded
     | ErrBalanceTxExistingCollateral
+    | ErrBalanceTxConflictingNetworks
     | ErrBalanceTxAssignRedeemers ErrAssignRedeemers
-    | ErrBalanceTxNotYetSupported BalanceTxNotSupportedReason
-    | ErrBalanceTxFailedBalancing Cardano.Value
+    | ErrBalanceTxInternalError ErrBalanceTxInternalError
+    | ErrBalanceTxZeroAdaOutput
```

Comment:
Interesting.

In the case of a zero-ada output, is this an ordinary user-specified output, or is it one that we've created during coin selection?

Reply:
> In the case of a zero-ada output, is this an ordinary user-specified output,

Yes.

Reply:
So we don't support users providing txs with 0 ada outputs — yet — because while coin-selection automatically increase their ada-values, we'd need 1) coin-selection to know about datums and other things affecting the minUTxOValue, 2) we'd need to update the outputs in balanceTx.